### PR TITLE
[stripe] Use attributionId in stripe queries in usage

### PR DESCRIPTION
**Pull request by easyCZ**
## Description

Customers in staging & prod have been udpated to additively contain `attributionId` metadata.

## Related Issue(s)

Fixes #

## Release Notes

```release-note
NONE
```

## Werft options:

- [ ] /werft with-preview

### CODE REVIEW

**Comment by roboquat** [comment]
@easyCZ: Adding the "do-not-merge/release-note-label-needed" label because no release-note block was detected, please follow our [release note process](https://git.k8s.io/community/contributors/guide/release-notes.md) to remove it.

<details>

Instructions for interacting with me using PR comments are available [here](https://git.k8s.io/community/contributors/guide/pull-requests.md).  If you have questions or suggestions related to my behavior, please file an issue against the [kubernetes/test-infra](https://github.com/kubernetes/test-infra/issues/new?title=Prow%20issue:) repository.
</details>

File: components/usage/pkg/stripe/stripe.go [hunk 1]
```diff
@@ -146,15 +149,15 @@ func (c *Client) updateUsageForCustomer(ctx context.Context, customer *stripe.Cu
...
+		Quantity:         stripe.Int64(credits),
 	})
 	if err != nil {
 		return nil, fmt.Errorf("failed to register usage for customer %q on subscription item %s", customer.Name, subscriptionItemId)
 	}
 
 	return &UsageRecord{
 		SubscriptionItemID: subscriptionItemId,
-		Quantity:           summary.Credits,
+		Quantity:           credits,
```
Comment (svenefftinge) [inline components/usage/pkg/stripe/stripe.go:160]: nit: do we use this struct anywhere? the name `UsageRecord` might be a bit confusing given we have records called `Usage`.

**Review by svenefftinge — APPROVED** [review]
Thanks! ❤️

File: components/usage/pkg/stripe/stripe.go (hunk 1 above)
Comment (easyCZ) [reply components/usage/pkg/stripe/stripe.go:160]: Fair. It was mostly to return something from method but we can easily return a tuple instead.

**Pull request merged**